testing/test_pathlib.py

```diff
@@ -413,6 +413,11 @@ def test_bestrelpath() -> None:
     assert bestrelpath(curdir, curdir.parent) == ".."
     assert bestrelpath(curdir, Path("hello")) == "hello"
 
+    # GH 9894
+    directory = Path("/var/tmp/pytest-issue")
```

Comment:
This test passes for me on Windows (which is fine), but @stphnlyd did you double check that this check fails without the rest of your changes?

Reply:
Yes with the change in pathlib.py both this test and my demo case in #9894 fail.